src/index.d.ts

```diff
@@ -1,7 +1,7 @@
 // Typescript definition file
 declare module 'fetch-intercept' {
   export interface FetchInterceptor {
-    request?(url: string, config: any): any[];
+    request?(url: string, config: any): Promise<any[]>;
```

Comment:
I think we need to allow both

```ts
request?(url: string, config: any): Promise<any[]> | any[];
```

Reply:
I think that is correct .. both sounds more proper

Reply:
Ok, just added the additional any[]